### PR TITLE
Refactor certain filters into its own service

### PR DESCRIPTION
**Pull request by Arif-Khalid**
### Summary:

Fixes part of #249 

#### Type of change:

- 🎨 Code Refactoring

### Changes Made:

* Refactored a few filters into their own service
* Does not break any behaviour for other filters
* Other filters will be refactored in another PR to reduce the size of this one
* Refactored following filters
  * Selected labels
  * Hidden labels
  * Status
  * Type

### Proposed Commit Message:

```
Filters are observed, subscribed and emitted in many different 
components.

This makes the code base hard to understand
and adapt for filtering in future components.

Let's refactor the filters into its own centralised service
```

<details><summary>
<h3>Checklist:</h3>
</summary>

- [x] I have tested my changes thoroughly.
- [x] I have created tests for any new code files created in this PR or provided a link to a issue/PR that addresses this.
- [x] I have added or modified code comments to improve code readability where necessary.
- [x] I have updated the project's documentation as necessary.

</details>

### CODE REVIEW

**Review by NereusWB922 — COMMENTED** [review]
LGTM! Tested the status, type, and label filters, and they are working.

**Review by NereusWB922 — APPROVED** [review]
LGTM!

File: src/app/core/services/filters.service.ts
Comment (vigneshsankariyer1234567890) [inline src/app/core/services/filters.service.ts:20]: Seems like there's duplication of the logic when updating dropdownFilter itself. For instance, `updateSelectedLabels` is in essence doing the same thing as `updateHiddenLabels` and even the other methods: to update the right property. Instead, maybe we could follow a pattern of composition using the `Partial<T>` API in TypeScript?

This could look something like this:

```ts
updateFilters(newFilters: Partial<DropdownFilter>): void {
  ...this.dropdownFilter$.value,
  ...newFilters,
}
```

Then after we update, we can check our rules for updating the status and type. 

Consumers would then simply pass some filters that they want to change in the components themselves.

File: src/app/shared/filter-bar/filter-bar.component.html
Comment (vigneshsankariyer1234567890) [inline src/app/shared/filter-bar/filter-bar.component.html:21]: As per my earlier comment, the change here would be something like

```html
<mat-select [value]="this.dropdownFilter.type" (selectionChange)="this.filtersService.updateFilters({type:$event.value})">
```

**Review by vigneshsankariyer1234567890 — CHANGES_REQUESTED** [review]
So far functionality seems ok! Just a small change requested to the Filters Service :)

File: src/app/core/services/filters.service.ts
Comment (Arif-Khalid) [reply src/app/core/services/filters.service.ts]: Agreed. This is a type I was not aware of, it is perfect in this situation.
I have made the appropriate changes.

**Review by gycgabriel — APPROVED** [review]
Looks neat, thanks for the refactor. Let's wait for vignesh approval

**Review by vigneshsankariyer1234567890 — APPROVED** [review]
LGTM 🔥 🚀  changes look great with the addition of the pipe!

**Pull request merged**
